stock_available_to_promise_release/models/product_product.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2020 Camptocamp (https://www.camptocamp.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    move_need_release_count = fields.Integer(
+        string="Moves Need Release", compute="_compute_move_need_release_count"
+    )
+
+    @api.depends("stock_move_ids.need_release")
+    def _compute_move_need_release_count(self):
+        for product in self:
+            product.move_need_release_count = len(
+                product.stock_move_ids.filtered("need_release")
+            )
```

Comment:
This would scale badly over time, with thousands of stock.moves per product

```suggestion
        data = self.env["stock.move"]._read_group(
            domain=[('product_id', 'in', self.ids), ('need_release', '=', True)],
            fields=['product_id'],
            groupby=['product_id'],
        )
        counts = {
            item['product_id'][0]: item['product_id_count']
            for item in data
        }
        for product in self:
            product.move_need_release_count = counts.get(product.id, 0)
```